net/portmapper/portmapper.go

```diff
@@ -409,10 +414,18 @@ func (c *Client) createOrGetMapping(ctx context.Context) (external netaddr.IPPor
 		}
 	}
 
-	// And ask for a mapping.
-	pmpReqMapping := buildPMPRequestMappingPacket(localPort, prevPort, pmpMapLifetimeSec)
-	if _, err := uc.WriteTo(pmpReqMapping, pmpAddru); err != nil {
-		return netaddr.IPPort{}, err
+	if haveRecentPMP {
+		// And ask for a mapping.
+		pmpReqMapping := buildPMPRequestMappingPacket(localPort, prevPort, pmpMapLifetimeSec)
```

Comment:
the scope of this variable is two lines, so this variable can be shorter. (and be the same as the ones on line 425/426 to match)

Maybe "pkt" to match your other caller of these above?